eclass/kernel-build.eclass

```diff
@@ -225,6 +233,34 @@ kernel-build_src_configure() {
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" olddefconfig
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" modules_prepare
 	cp -pR "${WORKDIR}"/modprep "${WORKDIR}"/build || die
+
+	# Now that we have a release file, set KV_FULL
+	local relfile=${WORKDIR}/build/include/config/kernel.release
+	if [[ -z ${KV_FULL} ]]; then
+		KV_FULL=$(<"${relfile}") || die
+	fi
+
+	# Make sure we are about to build the correct kernel
+	if [[ ${PV} != *9999 ]]; then
+		local expected_ver=$(dist-kernel_PV_to_KV "${PV}")
+		local expected_rel=$(<"${relfile}")
+
+		if [[ ${KV_FULL} != ${expected_rel} ]]; then
+			eerror "KV_FULL mismatch!"
+			eerror "KV_FULL:  ${KV_FULL}"
+			eerror "Expected: ${expected_rel}"
+			die "KV_FULL mismatch: got ${KV_FULL}, expected ${expected_rel}"
+		fi
+
+		if [[ ${KV_FULL} != ${expected_ver}* ]]; then
+			eerror "Kernel version does not match PV!"
+			eerror "Source version: ${KV_FULL}"
+			eerror "Expected (PV*): ${expected_ver}*"
+			eerror "Kernel version must match PV."
```

Comment:
This is redundant.

Reply:
It's not redundant, the checks are checking two different things to make sure that _everything_ is consistent.

The release file must match both KV_FULL _and_ PV for the prepared kernel sources to be valid for our tooling. We can't just check one and assume the other because they could still differ. We remove the first check, and then we're back to square one where our directory names can fall out of sync and break our tooling. We remove the second, and we can build arbitrary kernel versions with arbitrary PVs, which is undesirable from a package management perspective and a large regression from the current behaviour.

Reply:
No, I'm saying that saying "Kernel version must match PV" is redundant when it's also saying that it doesn't match PV. One implies the other.